npc/re/instances/InfiniteSpace.txt

```diff
@@ -0,0 +1,834 @@
+
+		if (.@floor == 50) {
+			mapannounce .@map$,"Reckless Explorer : It seems there are no other way from here. It's dangerous, so you'd better return outside.",0xFFFF00;
+			enablenpc instance_npcname("#w50_ppl");
+		}
+		if (.@floor < 50) {
+			if ((.@floor%10) == 0)
+				mapannounce .@map$,"Reckless Explorer : This area is safe. Use the passage to go to the next area.",0xFFFF00;
+			else
+				mapannounce .@map$,"Reckless Explorer : This area is safe. Go through the passage to the next area.",0xFFFF00;
+			// Enable warp portal to next floor
+			enablenpc instance_npcname("#w" + .@floor);
+			setd("'floor",.@floor + 1);
+		}
+	}
+OnPoringKill:
+	end;
+}
+
+1@infi,42,8,1	script	#w0_0	4_F_GON,{
+	.@mode = getd("'mode");
+	if (is_party_leader() && !.@mode) {
+		mes "[ Reckless Explorer ]";
+		mes "I have went in and out of this place many times and figured out a way to provoke the monsters in here. You can get more rewards if you're going to deal with heated ones.";
+		next;
+		mes "[ Reckless Explorer ]";
+		mes "What do you think? Would you like to try something challenging?";
+		next;
+		if (select("I want to proceed normally:I want to proceed with something difficult") == 1) {
+			setd("'mode",1);
+			setd("'poring_chance",1);
+		} else {
+			setd("'mode",2);
+			setd("'poring_chance",2);
+		}
+		mes "[ Reckless Explorer ]";
+		mes "Good. Shall we go now?";
+		close2;
+		disablenpc instance_npcname(strnpcinfo(0));
+		hideoffnpc instance_npcname("#w0");
+		npctalk "Reckless Explorer : Woah~ It's very neat for a sunken place, don't you think?",instance_npcname("#w0");
+		sleep 3000;
+		npctalk "Reckless Explorer : I would believe it to be an ancient site, not a lighthouse. It's so well organized...",instance_npcname("#w0");
+		sleep 3000;
+		npctalk "Reckless Explorer : Oh, did you hear something?",instance_npcname("#w0");
+		donpcevent instance_npcname("#inf_space_mobs") + "::OnSummon";
+		sleep 3000;
+		npctalk "Reckless Explorer : Oh, that must be the monsters haunting this place.",instance_npcname("#w0");
+		sleep 3000;
+		npctalk "Reckless Explorer : They look no different from the ones we saw outside, but I can feel a very strong energy.",instance_npcname("#w0");
+		sleep 3000;
+		npctalk "Reckless Explorer : Hm~ It looks dangerous, so I'll go back up. I'll give you directions from above.",instance_npcname("#w0");
+		disablenpc instance_npcname("#w0");
+	}
+	end;
+}
+
+1@infi,42,8,1	script	#w0	4_F_GON,{
+	end;
+}
+
+1@infi,366,392,4	script	#w50_ppl	4_F_GON,{
+	mes "[ Reckless Explorer ]";
+	mes "I'll keep a separate record of this expedition. If you've been lucky, you might have gotten a decent reward from your expedition.";
+	next;
+	mes "[ Reckless Explorer ]";
+	mes "So, shall we get out of here?";
+	close2;
+	warp "cmd_fild07",53,270;
+	end;
+}
+
+1@infi,30,369,3	script	#w10_t	4_TREASURE_BOX,{
+	if (is_party_leader()) {
+		specialeffect EF_COIN;
+		disablenpc instance_npcname(strnpcinfo(0));
+		sleep 1000;
```

Comment:
may be exploitable if the change of leader is allowed

Reply:
The npc is disabled eitherway. Can't see how that's exploitable unless packet spam to trigger it multiple times.
We could add a variable check or disable the npc before triggering the effect.

I don't have any replays with multiple players, i'm not even sure if only the leady can open them.

Reply:
No you're right I didn't see the disablenpc